File: AddOns/Share/Categories/TKSegment+Shareable.m
//
//  TKSegment+Shareable.m
//  Pods
//
//  Created by Adrian Schoenig on 24/06/2016.
//
//

#import "TKSegment+Shareable.h"

#import "TKShareHelper.h"

@implementation TKSegment (Shareable)

- (NSURL *)shareURL
{
  BOOL isEnd = (self.order == TKSegmentOrderingEnd);
  CLLocationCoordinate2D coordinate = isEnd ? [self.end coordinate] : [self.start coordinate];
  NSDate *time = isEnd ? self.arrivalTime : self.departureTime;
  return [TKShareHelper meetURLForCoordinate:coordinate atTime:time];
}

@end

File: AddOns/Share/Categories/TKSegment+Shareable.swift
//
//  TKSegment+Shareable.swift
//  TripKit
//
//  Created by Adrian Schoenig on 30/6/17.
//  Copyright © 2017 SkedGo. All rights reserved.
//

import Foundation

#if TK_NO_MODULE
#else
  import TripKit
#endif

extension TKSegment: TKURLShareable {
  public var shareURL: URL? {
    get {
      let isEnd = self.order() == .end
      guard let coordinate = isEnd ? end?.coordinate : start?.coordinate else { return nil }
      let time = isEnd ? arrivalTime : departureTime
      return TKShareHelper.createMeetURL(coordinate: coordinate, at: time)
    }
  }
  
}
